src/main/java/htsjdk/samtools/util/BinaryCodec.java

```diff
@@ -91,29 +93,41 @@
     //////////////////////////////////////////////////
 
     /**
-     * Constructs BinaryCodec from a file and set it's mode to writing or not
+     * Constructs BinaryCodec from a file and set its mode to writing or not
      *
-     * @param file    file to be written to or read from
+     * @param path    file to be written to or read from
      * @param writing whether the file is being written to
      */
-    public BinaryCodec(final File file, final boolean writing) {
+    public BinaryCodec(final Path path, final boolean writing) {
         this();
         try {
             this.isWriting = writing;
             if (this.isWriting) {
-                this.outputStream = IOUtil.maybeBufferOutputStream(new FileOutputStream(file));
-                this.outputFileName = file.getName();
+                this.outputStream = IOUtil.maybeBufferOutputStream(Files.newOutputStream(path));
+                this.outputFileName = path.toString();

```

Comment:
If we want to match the old behavior we should probably use `path.getFileName()` instead of `toString()`